api/python/provisioner/commands/setup_provisioner.py

```diff
@@ -1790,12 +1792,29 @@ def _run(self, nodes, **kwargs):  # noqa: C901 FIXME
         logger.info("Configuring provisioner logging")
         for state in [
             'components.system.prepare',
-            'components.provisioner.config'
         ]:
             ssh_client.cmd_run(
                 f"salt-call state.apply {state}",
                 targets=master_targets
             )
+        
+        # Seperation of variable to make flake8 happy
+        provisioner_pillar = {
+            'pillar': {
+                'provisioner': {
+                    'cluster': {
+                        'cluster_pillar_path':
+                            PRVSNR_PILLAR_DIR / "/groups/all/cluster.sls",
+                        'nodes': len(run_args.nodes)
+                    }
+                }
+            }
+        }
+        ssh_client.cmd_run(
+            f"salt-call state.apply components.provisioner.config",
+            targets=ALL_MINIONS,
```

Comment:
targeting al minions for that pillar destination in case of glusterfs sharing is overhead and even may relate to an issue.

I would suggest to make that more granular:
- split states appliance into:
  - ones that should be applied for all minions
  - ones for all masters only
  - ones that updates pillar:
     - in case of clustered (ha) we need to run that only on one node
     - in case of non-ha - all nodes